cmd/cephcsi.go

```diff
@@ -88,6 +88,8 @@ func init() {
 		"",
 		"list of Kubernetes node labels, that determines the"+
 			" CRUSH location the node belongs to, separated by ','")
+	flag.StringVar(&conf.ContainerOrchestrator, "container-orchestrator", "kubernetes",
+		"container orchestrator where cephcsi is running")
```

Comment:
It would be better to auto-detect this. You can check for the `KUBERNETES_SERVICE_HOST` environment variable, for example. When an other CO is supported, detection for that can be added.

A `config.GetContainerOrchestrator()` function can take care of the detection, and return a string constant for Kubernetes (in the `internal/util` package):

```go
type ContainerOrchestrator string

const (
    ContainerOrchestratorKubernetes = ContainerOrchestrator("kubernetes")
)

func (c *Config) GetContainerOrchestrator() ContainerOrchestrator {
    // do detection
    return ContainerOrchestratorKubernetes
}
```

Reply:
This will be out of scope for this PR and we can take up this is another PR?

Reply:
I think it would be better to do it in this PR, so that there is no need to add the `container-orchestrator` parameter, and remove it afterwards (breaks backwards compatibility).

Reply:
What about the default value for the flag? 
I was considering setting the flag's default value to an empty string. This way, if the user doesn't pass a value for the flag, we can automatically detect it. Does that sound good?

Reply:
Maybe add a `func RunsOnKubernetes() bool` to the `internal/util/k8s` package?